elastic/kibana
# [Lens] Rename datasources

## Summary

Closes https://github.com/elastic/kibana/issues/137723

This PR does the following things:
- Created a datasources folder and moved the two datasources inside
- Renamed the "textBasedLanguages" datasource to textBased (also renaming of types)
- Renamed the "indexPattern" datasource to formBased (also renaming of types)

Note1: I havent renamed css classes or data subjects. We could do it on a followup PR
Note2: If you have SOs with the text based language will fail because this PR also changes the id of this datasource. This is ok as we are introducing text based languages in Lens on this minor.
Note 3: This PR introduces a migration script.

### How to test
- Test that old by value and by reference Lens visualizations work as expected (ensure that migrations have run)
- Test that new Lens visualizations work as expected for the formBased datasource
- Test that new Lens visualizations work as expected for textBased datasource

The functional tests run successfully which is a great indicator that everything works as expected but I would appreciate a manual testing to ensure that everything works fine.
